src/FunctionApps/DSTP/src/main/kotlin/gov/cdc/prime/phdi/Utilities/HL7Parser.kt

```diff
@@ -0,0 +1,80 @@
+package gov.cdc.prime.phdi.Utilities
+
+import ca.uhn.hl7v2.DefaultHapiContext
+import ca.uhn.hl7v2.HL7Exception
+import ca.uhn.hl7v2.HapiContext
+import ca.uhn.hl7v2.model.Message
+import ca.uhn.hl7v2.parser.EncodingNotSupportedException
+import ca.uhn.hl7v2.parser.Parser
+import ca.uhn.hl7v2.validation.ValidationContext;
+import ca.uhn.hl7v2.validation.impl.ValidationContextFactory;
+
+public class HL7Parser() {
+    public fun parse(content: String): Map<String, MutableList<Any>> {
+        val rawMessages = convertBatchMessagesToList(content)
+        val processedMessages = mutableMapOf<String, MutableList<Any>>(
+            "invalid_messages" to mutableListOf(),
+            "valid_messages" to mutableListOf()
+        )
+
+        val context = DefaultHapiContext()
+        // We will likely replace this with a more custom validation class in the future
+        context.setValidationContext(ValidationContextFactory.defaultValidation() as ValidationContext);
+        val parser = context.getPipeParser()
+
+        rawMessages.forEach {
+            try {
+                val parsedMessage = parser.parse(it)
+                processedMessages["valid_messages"]!!.add(parsedMessage)
+            } catch (e: HL7Exception) {
+                processedMessages["invalid_messages"]!!.add(it)
+            }
+        }
+        return processedMessages
+    }
+
+    /* This method was adopted from PRIME ReportStream, which can be found here: 
+       https://github.com/CDCgov/prime-reportstream/blob/194396582be02fcc51295089f20b0c2b90e7c830/prime-router/src/main/kotlin/serializers/Hl7Serializer.kt#L121
+    */
+    public fun convertBatchMessagesToList(
```

Comment:
just a one sentence comment on what this is doing would be nice, even though it's copied from RS.